src/Map.ts

```diff
@@ -46,7 +46,7 @@ export const isMember = <A>(SA: Setoid<A>): (<K>(a: A, m: Map<K, A>) => boolean)
     let out: boolean = false
     const as = m.values()
     let e: IteratorResult<A>
-    while (!(e = as.next()).done) {
+    while (!(e = as.next()).done && !out) {
```

Comment:
I think that it could just `return` here

```ts
const as = m.values()
let e: IteratorResult<A>
while (!(e = as.next()).done) {
  if (SA.equals(a, e.value)) {
    return true
  }
}
return false
```